weather/weather.go

```diff
@@ -12,56 +10,51 @@ const (
 	weatherURL  = "https://api.openweathermap.org/data/2.5/weather?appid="
 )
 
-type Client struct {
+type WeatherClient struct {
 	apiKey string
 	unit   string
 
-	common service
-
-	Temperature *TemperatureService
-	Location    *LocationService
+	httpClient *resty.Client
 }
 
-type service struct {
-	client *Client
+type WeatherConfig struct {
+	ApiKey string
+	Units  string "metric" // This isn't setting a default
```

Comment:
I only gave you half the story here last time. In our code we are able to use struct tags to set defaults because we initialize our struct from environment variables. We use a library called `envconfig` that allows us to do that.